xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java

```diff
@@ -670,4 +671,12 @@ private static class FakeSocketAddress extends SocketAddress {
       return "FakeSocketAddress-" + name;
     }
   }
+
+  private static class FakeRandom extends Random {
+    @Override
+    public double nextDouble() {
+      setSeed(1L);
```

Comment:
You need to set the seed in a constructor rather than each time you get a new value.  Otherwise you'll keep getting the same value.

Reply:
I do want to make it get the same value - this basically disables the initial randomization. I did that because some tests are unhappy.

There are tests that enables it - it works because those tests runs 1000 times.

Reply:
Then just return a constant. No need to call `setSeed()`.

Reply:
If some tests are unhappy, you should fix the tests.  What you have is the same as
  public double nextDouble() {
    return 0.7308781907032909;
  }
It doesn't matter how many times it is run and isn't really testing the randomness you want.  

I think what you really want is to pass it a list of numbers to return and when it gets to the end it just starts over.  That would give you the flexibility to really test that different weights and different orderings are being handled correctly.

Reply:
Determination is necessary to evaluate certain behavior, e.g. earlier deadline is picked first, same weight items order breaks the tie.  So I intentionally inject fake random there.
Different weights with randomization is tested in, e.g. `edfScheduler` test case.